src/Serialization/Compact.php

```diff
@@ -41,13 +41,19 @@ public function __construct(
 
     /**
      * @param string $headersJson
+     *
      * @return HeaderParameter\ParameterInterface[]
+     * @throws \InvalidArgumentException
      */
     protected function parseHeaders($headersJson)
     {
         $parameters = [];
         $headers = json_decode($headersJson, true);
 
+        if (!is_array($headers) || empty($headers)) {
```

Comment:
@emarref Yeah, it would be useful. The parsing _error message_ can be received from a `json_last_error_msg` method, but it's not yet available in the minimum supported PHP version of the library (5.4), only from 5.5+. Does it make sense to do backporting of this method?